src/model/Model.js

```diff
@@ -261,7 +267,7 @@ export default class Model extends ModelBase {
         const attr = jsonAttr[i];
         const value = json[attr];
 
-        if (_.isObject(value)) {
+        if (_.isObject(value) && !(value instanceof KnexRaw)) {
           json[attr] = JSON.stringify(value);
```

Comment:
Yes. I added test case, where json column value is updated with knex.raw and it fails even without reference builder stuff. 

This failed because circular data structure:
```js
        return BoundModel.query()
          .update({
            jsonArray: BoundModel.knex().raw('to_jsonb(??)', ['name'])
          }).then(function (result) {
            expect(result).to.be(4);
          });
```